### PR TITLE
feature: add privileged_agent connections.

### PR DESCRIPTION
**Pull request by yaolaiyaoqu**
add privileged_agent connections to set cycle connections

### CODE REVIEW

File: patches/nginx-1.19.3-privileged_agent_process.patch [hunk 1]
```diff
@@ -196,7 +200,7 @@ index 7cee1c5..c4f70d6 100644
 +static void
  ngx_cache_manager_process_handler(ngx_event_t *ev)
  {
-     time_t        next, n;
+     ngx_uint_t    i;
```
Comment (doujiang24) [inline patches/nginx-1.19.3-privileged_agent_process.patch:203]: the nginx 1.19.3 codebase has changed, so we have this change.

File: patches/nginx-1.19.3-privileged_agent_process.patch [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
+index f73e559..949a8d0 100644
 --- a/src/core/nginx.c
 +++ b/src/core/nginx.c
-@@ -981,6 +981,7 @@ ngx_core_module_create_conf(ngx_cycle_t *cycle)
+@@ -1028,6 +1028,8 @@ ngx_core_module_create_conf(ngx_cycle_t *cycle)
  
      ccf->daemon = NGX_CONF_UNSET;
      ccf->master = NGX_CONF_UNSET;
-+    ccf->privileged_agent = NGX_CONF_UNSET;
++    ccf->privileged_agent             = NGX_CONF_UNSET;
```
Comment (doujiang24) [inline patches/nginx-1.19.3-privileged_agent_process.patch:9]: style: we don't need to change this line.

File: patches/nginx-1.19.9-privileged_agent_process.patch [hunk 3]
```diff
@@ -188,7 +192,7 @@ index 7cee1c5..c4f70d6 100644
 +static void
  ngx_cache_manager_process_handler(ngx_event_t *ev)
  {
-     time_t        next, n;
+     ngx_uint_t    i;
```
Comment (zhuizhuhaomeng) [inline patches/nginx-1.19.9-privileged_agent_process.patch:195]: why change this ?

File: patches/nginx-1.19.9-privileged_agent_process.patch [hunk 4]
```diff
@@ -32,16 +34,17 @@ index c51b7ff..3261f90 100644
...
+@@ -89,6 +92,8 @@ struct ngx_cycle_s {
  typedef struct {
      ngx_flag_t                daemon;
      ngx_flag_t                master;
 +    ngx_flag_t                privileged_agent;
++    ngx_uint_t                privileged_agent_connections;
  
      ngx_msec_t                timer_resolution;
- 
+     ngx_msec_t                shutdown_timeout;
```
Comment (zhuizhuhaomeng) [inline patches/nginx-1.19.9-privileged_agent_process.patch:45]: why add this？

**Review by zhuizhuhaomeng — CHANGES_REQUESTED** [review]

File: patches/nginx-1.19.9-privileged_agent_process.patch (hunk 3 above)
Comment (yaolaiyaoqu) [reply patches/nginx-1.19.9-privileged_agent_process.patch:195]: may be the previous patch is not base on 1.19.9

**Review by doujiang24 — APPROVED** [review]

**Review by zhuizhuhaomeng — APPROVED** [review]

**Pull request merged**